app/src/modules/settings/routes/data-model/collections/components/collection-item.vue

```diff
@@ -10,23 +10,69 @@ import { CollectionTree } from '../collections.vue';
 const props = defineProps<{
 	collection: Collection;
 	collections: Collection[];
+	expandAll?: boolean;
+	forceUpdate?: boolean;
 	visibilityTree: CollectionTree;
 	disableDrag?: boolean;
 }>();
 
 const emit = defineEmits(['setNestedSort', 'editCollection']);
 
 const { t } = useI18n();
-const { data: isCollectionExpanded } = useLocalStorage(`settings-collapsed-${props.collection.collection}`, true);
+
+const { data: isCollectionExpanded } = useLocalStorage(
+	`settings-collapsed-${props.collection.collection}`,
+	true as boolean,
+);
 
 const toggleCollapse = () => {
 	isCollectionExpanded.value = !isCollectionExpanded.value;
 };
 
+watch(
+	() => props.forceUpdate,
+	() => {
+		isCollectionExpanded.value = props.expandAll;
+	},
+);
+
 const nestedCollections = computed(() =>
 	props.collections.filter((collection) => collection.meta?.group === props.collection.collection),
 );
 
+watch(
+	() => props.expandAll,
+	(expanded) => {
+		switch (expanded) {
+			case true:
+				if (nestedCollections.value.length > 0) {
+					if (props.collection.meta?.group) {
+						isCollectionExpanded.value = expanded;
+					} else {
+						return;
+					}
+				} else {
+					isCollectionExpanded.value = expanded;
+				}
+
+				break;
+			case false:
+				if (nestedCollections.value.length > 0) {
+					if (props.collection.meta?.group) {
+						isCollectionExpanded.value = expanded;
+					} else {
+						return;
+					}
+				} else {
+					isCollectionExpanded.value = expanded;
+				}
+
+				break;
+		}
+	},
+	{ immediate: true },
+);
+
```

Comment:
This approach is not very clean 😅, but I wanted to first confirm if this meets your expectations before refactoring.

Reply:
I have an updated strategy below 🙂 

Other than that, an `if/else` statement is probably better suited than a `switch` for a simple yes/no check